qemu/tests/live_snapshot_basic.py

```diff
@@ -0,0 +1,106 @@
+        self.env = env
+        self.vm = env.get_vm(params["main_vm"])
+        self.snapshot_file = params.get("snapshot_file")
+        self.snapshot_format = params.get("snapshot_format", "qcow2")
+        self.snapshot_mode = params.get("snapshot_mode", "absolute-paths")
+        self.node_name = params.get("node_name")
+        self.snapshot_node_name = params.get("snapshot_node_name")
+
+    def create_image(self):
+        """
+        Create a image.
+        """
+        image_name = self.params.get("image_name")
+        self.params['image_name_snapshot'] = image_name + "-snap"
+        image_name_snapshot = self.params['image_name_snapshot']
+        snapshot_params = self.params.object_params("snapshot")
+        base_dir = self.params.get("images_base_dir", data_dir.get_data_dir())
+
+        image_io = QemuImg(snapshot_params, base_dir, image_name)
+        image_name, _ = image_io.create(snapshot_params)
+        return image_name
+
+    def get_base_image(self):
+        """
+        Get base image.
+        """
+        block_info = self.vm.monitor.info("block")
+        if self.vm.monitor.protocol == 'qmp':
+            device = block_info[0]["device"]
+        else:
+            device = "".join(block_info).split(":")[0]
+        return device
+
+    def create_snapshot(self):
+        """
+        Create a live disk snapshot.
+        """
+        if self.snapshot_mode == "existing":
+            error_context.context("Creating an image ...", logging.info)
+            self.snapshot_file = self.create_image()
+        device = self.get_base_image()
+        if self.node_name and self.snapshot_node_name:
+            self.vm.monitor.live_snapshot(device, self.snapshot_file,
+                                          format=self.snapshot_format,
+                                          mode=self.snapshot_mode,
+                                          node_name=self.node_name,
+                                          snapshot_node_name=self.snapshot_node_name)
+        else:
+            self.vm.monitor.live_snapshot(device, self.snapshot_file,
+                                          format=self.snapshot_format,
+                                          mode=self.snapshot_mode)
+
+    def check_snapshot(self):
+        """
+        Check whether the snapshot is created successfully.
+        """
+        snapshot_info = str(self.vm.monitor.info("block"))
+        if self.snapshot_file not in snapshot_info:
+            logging.error(snapshot_info)
+            raise exceptions.TestFail("Snapshot doesn't exist")
+        if self.snapshot_node_name and (self.snapshot_node_name not in
+                                        snapshot_info):
+            logging.error(snapshot_info)
+            raise exceptions.TestFail("There is no node name for snapshot")
+
+
+@error_context.context_aware
+def run(test, params, env):
+    vm = env.get_vm(params["main_vm"])
+    snapshot_test = LiveSnapshotBase(params, env)
```

Comment:
better to refactor test params via image tag before pass it to your LiveSnapshotBaseTest class
eg, 
snapshot_params = param.object_params(image_tag)